packages/stencil/components/progress-bar/src/components/osds-progress-bar/osds-progress-bar.tsx

```diff
@@ -0,0 +1,57 @@
+import { Component, Element, Host, h, Prop, Watch } from '@stencil/core';
+import {
+  OdsProgressBar,
+  OdsProgressBarController, odsProgressBarDefaultAttributes,
+  OdsProgressBarEvents,
+  OdsProgressBarMethods,
+} from '@ovhcloud/ods-core';
+import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';
+
+/**
+ * @slot (unnamed) - ProgressBar content
+ */
+@Component({
+  tag: 'osds-progress-bar',
+  styleUrl: 'osds-progress-bar.scss',
+  shadow: true
+})
+export class OsdsProgressBar implements OdsProgressBar<OdsStencilMethods<OdsProgressBarMethods>, OdsStencilEvents<OdsProgressBarEvents>> {
+  controller: OdsProgressBarController = new OdsProgressBarController(this);
+  @Element() el!: HTMLElement;
+
+  /** @see OdsProgressBarAttributes.max */
+  @Prop({ reflect: true }) public max?: OdsProgressBar['max'] = odsProgressBarDefaultAttributes.max;
+
+  /** @see OdsProgressBarAttributes.min */
+  @Prop({ reflect: true }) public min?: OdsProgressBar['min'] = odsProgressBarDefaultAttributes.min;
+
+  /** @see OdsProgressBarAttributes.progress */
+  @Prop({ reflect: true }) public progress?: OdsProgressBar['progress'] = odsProgressBarDefaultAttributes.progress;
+
+  /**
+   * @see OdsProgressBarBehavior.beforeRender
+   */
+  componentDidLoad(): void {
```

Comment:
maybe better to use `componentWillLoad` as it is called before rendering whereas `componentDidLoad` is called after rendering, thus a second rendering with the `defineProgressCssValue` call.